File: server/models/event.js
const mongoose = require("mongoose")

const EventSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
  },
  date: {
    type: String,
    required: true,
  },
  time: {
    type: String,
    required: true,
  },
  location: {
    type: String,
    required: true,
  },
  description: {
    type: String,
  },
  imageUrl: {
    type: String,
  },
  organizer: {
    type: String,
    required: true,
  },
  attendees: [
    {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
    },
  ],
  createdAt: {
    type: Date,
    default: Date.now,
  },
})

module.exports = mongoose.model("Event", EventSchema)

File: lib/models/event.dart
class Event {
  final String id;
  final String name;
  final String date;
  final String time;
  final String location;
  final String? description;
  final String? imageUrl;
  final String organizer;

  Event({
    required this.id,
    required this.name,
    required this.date,
    required this.time,
    required this.location,
    this.description,
    this.imageUrl,
    required this.organizer,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'],
      name: json['name'],
      date: json['date'],
      time: json['time'],
      location: json['location'],
      description: json['description'],
      imageUrl: json['imageUrl'],
      organizer: json['organizer'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'date': date,
      'time': time,
      'location': location,
      'description': description,
      'imageUrl': imageUrl,
      'organizer': organizer,
    };
  }
}

